dotnet/src/VectorDataIntegrationTests/MongoDBIntegrationTests/Support/MongoDBTestStore.cs

```diff
@@ -12,19 +11,18 @@ internal sealed class MongoDBTestStore : TestStore
 {
     public static MongoDBTestStore Instance { get; } = new();
 
+#pragma warning disable CA2213 // Disposable fields should be disposed
     private readonly MongoDbContainer _container = new MongoDbBuilder()
```

Comment:
The container does not implement `IDisposable`, but it implements `IAsyncDisposable` (not part of Full Framework) and I was getting another warning "disposable fields should be disposed".

Since `IAsyncDisposable` is not available for Full Framework (IIRC it was added to the `Microsoft.Bcl.AsyncInterfaces` package and it's being closed as part of the `StopAsync` method, I decided to simply ignore the warning.

The alternative would be to add dependency to `Microsoft.Bcl.AsyncInterfaces` and implement `IAsyncDisposable` (and mix sync and async code since we don't provide `IAsyncDisposable` implementation for store and collection)

Reply:
Hmm, interesting. For the tests we obviously don't care too much.

If we do want to at some point manage MongoDbContainer for the user (not just in the tests), we could go all the way and also implement IAsyncDisposable on VectorStore/VectorStoreCollection; aside from this we also have NpgsqlDataSource which implements IAsyncDisposable (though I can say that in practice it doesn't matter - at least not at the moment). I think Full Framework isn't a problem - adding a reference Microsoft.Bcl.AsyncInterfaces from MEVD seems reasonable...

But in any case, as discussed now that we have abstract base classes, we can do this in the future without breaking.